Discovery/discoveryplugin.py

```diff
@@ -362,11 +381,13 @@ def read_config(self):
 
 
     def show_config_dialog(self):
+        dlg = config_dialog.ConfigDialog(self.config_combo)
+        if (dlg.config_combo.currentIndex() >= 0):
+            dlg.configOptions.setCurrentIndex(dlg.config_combo.currentIndex())
 
-        dlg = config_dialog.ConfigDialog()
         if dlg.exec_():
             dlg.write_config()
-            self.read_config()
+            self.config_combo.setCurrentIndex(dlg.configOptions.currentIndex())
```

Comment:
Bug: after change in the current config the configuration is not refreshed - I need to switch to a different config and back to get it reloaded.